pkg/webhook/default_server/uniteddeployment/validating/uniteddeployment_validation.go

```diff
@@ -0,0 +1,166 @@
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("topology", "subset", "name"), subset.Name, fmt.Sprintf("invalid subset name %s", strings.Join(errs, ", "))))
+		}
+
+		if spec.Replicas == nil || subset.Replicas == nil {
+			continue
+		}
+
+		replicas, err := parseUnitReplicas(*spec.Replicas, *subset.Replicas)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("topology", "subset", "replicas"), subset.Replicas, fmt.Sprintf("invalid replicas %s", subset.Replicas.String())))
+		} else {
+			sumReplicas += replicas
+			count++
+		}
+	}
+
+	// summary of subset replicas may be less than uniteddployment replicas
+	if sumReplicas > *spec.Replicas {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("topology", "subset"), sumReplicas, fmt.Sprintf("summary of indicated subset replicas %d should not be greater than UnitedDeployment replicas %d", sumReplicas, *spec.Replicas)))
+	}
+
+	if count > 0 && count == len(spec.Topology.Subsets) && sumReplicas != *spec.Replicas {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("topology", "subset"), sumReplicas, fmt.Sprintf("if replicas of all subsets are provided, the summary of indicated subset replicas %d should equals UnitedDeployment replicas %d", sumReplicas, *spec.Replicas)))
+	}
+
+	return allErrs
+}
+
+func parseUnitReplicas(replicas int32, unitReplicas intstr.IntOrString) (int32, error) {
+	if unitReplicas.Type == intstr.Int {
+		if unitReplicas.IntVal < 0 {
+			return 0, fmt.Errorf("unitReplicas (%d) should not be less than 0", unitReplicas.IntVal)
+		}
+		return unitReplicas.IntVal, nil
+	}
+
+	strVal := unitReplicas.StrVal
+	if !strings.HasSuffix(strVal, "%") {
+		return 0, fmt.Errorf("unitReplicas (%s) only support int value or percentage value with a suffix '%%'", strVal)
+	}
+
+	intPart := strVal[:len(strVal)-1]
+	percent64, err := strconv.ParseInt(intPart, 10, 32)
+	if err != nil {
+		return 0, fmt.Errorf("unitReplicas (%s) should be correct percentage value", strVal)
+	}
+
+	if percent64 > int64(100) || percent64 < int64(0) {
+		return 0, fmt.Errorf("unitReplicas (%s) should be in range (0, 100]", strVal)
+	}
+
+	return int32(round(float64(replicas) * float64(percent64) / 100)), nil
+}
+
+func round(x float64) int {
+	return int(math.Floor(x + 0.5))
+}
+
+// validateUnitedDeployment validates a UnitedDeployment.
+func validateUnitedDeployment(unitedDeployment *appsv1alpha1.UnitedDeployment) field.ErrorList {
+	allErrs := apivalidation.ValidateObjectMeta(&unitedDeployment.ObjectMeta, true, apimachineryvalidation.NameIsDNSSubdomain, field.NewPath("metadata"))
+	allErrs = append(allErrs, validateUnitedDeploymentSpec(&unitedDeployment.Spec, field.NewPath("spec"))...)
+	return allErrs
+}
+
+// ValidateUnitedDeploymentUpdate tests if required fields in the UnitedDeployment are set.
+func ValidateUnitedDeploymentUpdate(unitedDeployment, oldUnitedDeployment *appsv1alpha1.UnitedDeployment) field.ErrorList {
```

Comment:
TODO: We need make checks. It seems that only a few fields such as spec.replica, subset.replica and partition are allowed to be changed. Done in future changes, of course.